ml-agents/mlagents/trainers/learn.py

```diff
@@ -125,14 +125,15 @@ def run_training(run_seed: int, options: RunOptions) -> None:
         )
 
         trainer_factory = TrainerFactory(
-            options.behaviors,
-            write_path,
-            not checkpoint_settings.inference,
-            checkpoint_settings.resume,
-            run_seed,
-            env_parameter_manager,
-            maybe_init_path,
-            False,
+            trainer_config=options.behaviors,
+            output_path=write_path,
+            train_model=not checkpoint_settings.inference,
+            load_model=checkpoint_settings.resume,
+            seed=run_seed,
+            param_manager=env_parameter_manager,
+            init_path=maybe_init_path,
+            multi_gpu=False,
+            use_torch="torch" in DetectDefault.non_default_args,
```

Comment:
maybe call this `force_torch`? since it is still possible for this to be `False` but `torch` to be specified in the YAML (and it will still use PyTorch).